src/org/exist/storage/XQueryPool.java

```diff
@@ -160,78 +155,57 @@ private synchronized void returnObject(final Source source, final CompiledXQuery
         }
     }
 
-    private Object borrowObject(final DBBroker broker, final Source source) {
-        final Source key;
-        final CompiledXQuery query;
-        synchronized (this) {
-            final int idx = getIndex(source);
-            if (idx < 0) {
-                return null;
-            }
-            key = (Source) keys[idx];
-            int validity = key.isValid(broker);
-            if (validity == Source.UNKNOWN) {
-                validity = key.isValid(source);
-            }
+    private Object borrowObject(final Source source) {
```

Comment:
@adamretter good for whom? I, as author of this changes, believe that it must stay as it because it clearly define each step of modification.

Also, "keep clean" history of 17k commits sound like paranoia for me (personally).